context/app/static/js/helpers/multiFetcher.ts

```diff
@@ -1,5 +1,21 @@
 // Fetcher function that lets SWR fetch multiple urls at once
 export const multiFetcher = (...urls: string[]) => {
   const f = (url: string) => fetch(url).then((response) => response.json());
+  if (urls.length === 0) {
+    return Promise.resolve([]);
+  }
+  return Promise.all(urls.map((url) => f(url)));
+};
+
+export const multiFetcherWithToken = (token: string, ...urls: string[]) => {
+  const f = (url: string) =>
+    fetch(url, {
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+    }).then((response) => response.json());
+  if (urls.length === 0) {
+    return Promise.resolve([]);
+  }
```

Comment:
Could we refactor out the commonalities and enable the `multiFetcher` to accept fetch options as a an arg with a default value of an empty obj?

Reply:
- `fetcher` and `multiFetcher` moved to a single `swr` file in helpers
- Both functions take in an optional `requestInit` object
- `fetcher` uses `multiFetcher` to minimize duplication
- Added JSDoc explanations for usage which appear on hover in VS Code


![image](https://github.com/hubmapconsortium/portal-ui/assets/19957804/d392dec5-ef0b-4af9-8ec0-1d6812356907)

![image](https://github.com/hubmapconsortium/portal-ui/assets/19957804/1ee2cf7e-6cf9-4267-a493-d6c112512972)